src/fluree/db/conn/remote.cljc

```diff
@@ -0,0 +1,156 @@
+
+(defn remote-lookup
+  [state servers ledger-address]
+  (go-try
+    (let [head-commit  (<? (remote-read state servers ledger-address false))
+          head-address (get head-commit "address")]
+      head-address)))
+
+;; NOTE, below function works in conjunction with message broadcasting (not in current PR)
+#_(defn remote-lookup
+  [state servers ledger-address]
+  (go-try
+    (or (get-in @state [:lookup ledger-address])
+          (let [head-commit  (<? (remote-read state servers ledger-address false))
+                head-address (get head-commit "address")]
+            (swap! state assoc-in [:lookup ledger-address] head-address)
+            (swap! state assoc-in [:resource head-address] head-commit)
+            head-address))))
+
+(defn remote-ledger-exists?
+  [state servers ledger-address]
+  (go-try
+    (boolean
+      (<? (remote-lookup state servers ledger-address)))))
+
+
+(defrecord RemoteConnection [id servers state lru-cache-atom serializer
+                             ledger-defaults parallelism msg-in-ch msg-out-ch]
+
+  conn-proto/iStorage
+  (-c-read [_ commit-key] (remote-read state servers commit-key false))
+  (-ctx-read [_ context-key] (remote-read state servers context-key false))
+  (-index-file-read [_ index-address] (remote-read state servers index-address true))
+
+  conn-proto/iNameService
+  (-pull [this ledger] :TODO)
+  (-subscribe [this ledger] :TODO)
+  (-alias [this address]
+    address)
+  (-lookup [this ledger-alias]
+    (remote-lookup state servers ledger-alias))
+  (-address [_ ledger-alias {:keys [branch] :or {branch :main} :as _opts}]
+    (go (str ledger-alias "/" (name branch) "/head")))
+  (-exists? [_ ledger-address]
+    (remote-ledger-exists? state servers ledger-address))
+
+  conn-proto/iConnection
+  (-close [_]
+    (log/info "Closing memory connection" id)
+    (swap! state assoc :closed? true))
+  (-closed? [_] (boolean (:closed? @state)))
+  (-method [_] :remote)
+  (-parallelism [_] parallelism)
+  (-id [_] id)
+  (-default-context [_] (:context ledger-defaults))
+  (-did [_] (:did ledger-defaults))
+  (-msg-in [_ msg] (go-try
+                     ;; TODO - push into state machine
+                     (log/warn "-msg-in: " msg)
+                     :TODO))
+  (-msg-out [_ msg] (go-try
+                      ;; TODO - register/submit event
+                      (log/warn "-msg-out: " msg)
+                      :TODO))
+  (-state [_] @state)
+  (-state [_ ledger] (get @state ledger))
+
+  index/Resolver
+  (resolve
+    [conn {:keys [id leaf tempid] :as node}]
+    (let [cache-key [::resolve id tempid]]
+      (if (= :empty id)
+        (storage/resolve-empty-node node)
+        (conn-cache/lru-lookup
+          lru-cache-atom
+          cache-key
+          (fn [_]
+            (storage/resolve-index-node conn node
+                                        (fn [] (conn-cache/lru-evict lru-cache-atom cache-key))))))))
+
+  #?@(:clj
```

Comment:
I think we can just not implement the protocol here instead of implementing it and throwing an unsupported exception on the only method.